activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -222,7 +235,7 @@ def encode_with(coder)
     end
 
     UNSAFE_STRING_METHODS.each do |unsafe_method|
-      if unsafe_method.respond_to?(unsafe_method)
+      if String.method_defined?(unsafe_method)
```

Comment:
Speed is not a problem here. This loop only run once, `UNSAFE_STRING_METHODS` times, so I prefer to keep the `respond_to?`